2020-an-environment-for-sustainable-research-software-in-germany-and-beyond/manuscript.md

```diff
@@ -0,0 +1,2021 @@
+performance and design of the software. (3) The use of the software in
+publications is a measure of impact but quantitative assessment brings
+about additional challenges (Li, Chen, and Yan 2019). Therefore, other,
+potentially domain-specific, impact measures, such as influence on
+policy and practice as well as use in other software and products should
+be considered as well to evaluate relevance for academia and society.
+Considerable attendance at training and networking events can be
+considered as a proof of use as well. (4) A **market analysis** needs to
+show that the software is important to a user base of relevant size and
+either unique or one of the main players in a field with several
+existing solutions. Geographical or political aspects can be considered
+as well, e.g. to support the maintenance of a European solution. A
+convergence process of (parts of) a research community towards a
+specific software stack, i.e. documented transition of several research
+groups to a common software, would be a strong indicator of impact.
+(5) As community benefits are a central goal of sustained software
+funding, outreach and **appropriate training material** for new users of
+the software are essential.
+
+### Software Quality
+
+As mandatory criteria of software quality that have to be fulfilled, we
+consider (6) the **public availability of the source code**, with
+(7) **version control** (ideally on a public platform).
+(8) **Documentation** of the software needs to be publicly available
+comprising both user documentation (requirements, installation, getting
+started, user manual) and developer documentation (ideally within the
+code, e.g. using Doxygen) (Lee 2018). (9) The **license** under which
+the software is distributed must be defined and ensure public code
+availability and free use, at least for academic purposes.
+(10) **Dependencies** on libraries and technologies must be defined.
+
+We acknowledge that some additional criteria have to be evaluated under
+consideration of the research domain. These comprise (11) the
+availability of examples (comprising input data and reference results),
+(12) mechanisms for extensibility (software modularity) as one aspect of
+software architecture (Venters et al. 2018) and (13) interoperability
+(APIs / common and open data formats for input and output), (14) testing
+infrastructure including but not strictly requiring unit tests,
+performance tests, regression tests, integration tests, end-to-end tests
+(ideally run in an automated fashion in a continuous integration
+environment), (15) tagged releases (considering their frequency, and
+availability for end users in terms of binary packages for major
+operating systems, or availability via package managers or containers),
+(16) no (large-scale) re-implementations for functionality for which
+good solutions already exist. Many of these aspects require appropriate
+infrastructure (see page ).
+
```

Comment:
An often overlooked but nevertheless important aspect is the communication between users and developers of a software: As the software matures and gets new releases, the users have to be notified of these updates. The users on the other hand have to be able to easily reach out to the development team with suggestions and issue reports. Even in a small environment, like an office internal software project, the expectations from the users and the plans from the development team easily go in different directions, if issues and progress are not being tracked in a transparent manner like an issue tracking system.

Reply:
I've integrated this aspect below in the the Maturity section